kernel/src/table.rs

```diff
@@ -76,6 +77,46 @@ impl Table {
         &self.location
     }
 
+    /// Creates a [`LogHistoryManager`] from an existing [`Snapshot`].
+    ///
+    /// This method allows reusing an existing snapshot to create a history manager
+    /// that can resolve timestamps for commits up to and including `snapshot.version()`.
+    ///
+    /// # Parameters
+    /// - `engine`: The [`Engine`] implementation to use for data access
+    /// - `snapshot`: An existing snapshot to initialize the history manager with
+    /// - `limit`: Optional maximum number of versions to track. When specified, the earliest
+    ///   queryable version will be `snapshot.version() - limit`. This allows trading
+    ///   memory usage for historical reach.
+    pub fn history_manager_from_snapshot(
```

Comment:
considering we have two constructors here already and we have optional info in each of them, would it make more sense to just have a builder?
```rust
let hm = HistoryManagerBuilder::new(engine)
    .withSnapshot(snapshot)
    .withLimit(1_000)
    .build()
```

aside: I wonder if it's more idiomatic to have engine in `new` or `build`..?

and should we include end version?

Reply:
Now all functions are static, so we avoid this :)